asf_public_discourse_home_decarbonisation/analysis/mse/eda_mse_category_data.py

```diff
@@ -0,0 +1,312 @@
+            logger.info(
+                "All usernames corresponding to multiple user_ids are Deleted Users."
+            )
+        else:
+            logger.warning(
+                "There are usernames corresponding to multiple user_ids that are not Deleted Users."
+            )
+
+    check_id_username2 = mse_data.groupby("user_id")[["username"]].nunique()
+
+    ids_corresponding_to_multiple_usernames = check_id_username2[
+        check_id_username2["username"] > 1
+    ]
+    if len(ids_corresponding_to_multiple_usernames) > 0:
+        logger.warning(
+            f"There are {len(ids_corresponding_to_multiple_usernames)} user IDs corresponding to multiple usernames."
+        )
+
+
+def plot_number_interactions_per_user(mse_data: pd.DataFrame, category: str):
+    """
+    Plots and saves a histogram with number of interactions per user.
+    The computation is based on all MSE posts, not necessarily the ones we collected data for.
+
+    Args:
+        mse_data (pd.DataFrame): Money Saving Expert data
+        cateogry (str): MSE category/sub-forum
+    """
+
+    interactions_per_user_overall = mse_data[["user_id", "n_posts_user"]]
+
+    # due to data collection at different times, numbers might differ, so we need to remove duplicates for each user
+    interactions_per_user_overall.sort_values("n_posts_user", ascending=False)
+    interactions_per_user_overall.drop_duplicates("user_id", keep="first", inplace=True)
+
+    plt.figure(figsize=(pc.figsize_x, pc.figsize_y))
+    plt.hist(
+        interactions_per_user_overall["n_posts_user"],
+        bins=30,
+        color=pc.NESTA_COLOURS[0],
+        edgecolor="white",
+    )
+    plt.xlabel("Number of posts and replies")
+    plt.ylabel("Number of users")
+    plt.title("Number of users with a given\nnumber of posts and replies")
+    plt.yscale("log")
+    plt.tight_layout()
+    path_to_plot = f"{MSE_FIGURES_PATH}/category_{category}_interactions_per_user.png"
+    plt.savefig(path_to_plot)
+    logger.info(
+        f"Plot with number of interactions per user for category '{category}' saved successfully in {path_to_plot}"
+    )
+
+
+def plot_number_posts_per_user(mse_data: pd.DataFrame, category: str):
+    """
+    Plots and saves a histogram with number of posts per user.
+    The computation is based on data collected (i.e. posts collected from the specified category,
+    not the whole MSE)
+
+    Args:
+        mse_data (pd.DataFrame): Money Saving Expert data
+    """
+    posts_per_user_data_collected = (
+        mse_data[mse_data["is_original_post"] == 1].groupby("user_id")[["id"]].nunique()
+    )
+
+    posts_per_user_data_collected.reset_index(inplace=True)
+    posts_per_user_data_collected.columns = ["user", "n_posts_user"]
+
+    plt.figure(figsize=(pc.figsize_x, pc.figsize_y))
+    plt.hist(
+        posts_per_user_data_collected["n_posts_user"],
+        bins=range(0, 90, 5),
```

Comment:
Notice the range is the same no matter the category, I think this is nice for comparison, I just guess if you have a small enough category (e.g. like "lpg-heating-oil-solid-other-fuels") there is a bit of white space. But I think I might be too nit-picky here to be frank :).

Reply:
not nit picky at all @helloaidank. glad you caught this, it was a lapse on my side. If we get data w/ values above 90, then this would actually be wrong, so I'll need to fix it. thanks for pointing it out! :)

Reply:
Fixed @helloaidank. I used `range(0, posts_per_user_data_collected["n_posts_user"].max()+5, 5),` instead.